dev/docker-entrypoint.sh

```diff
@@ -25,40 +25,75 @@ if [ "$1" = 'couchdb' ]; then
 fi
 
 if [ "$1" = '/opt/couchdb/bin/couchdb' ]; then
-	# Check that we own everything in /opt/couchdb and fix if necessary. We also
-	# add the `-f` flag in all the following invocations because there may be
-	# cases where some of these ownership and permissions issues are non-fatal
-	# (e.g. a config file owned by root with o+r is actually fine), and we don't
-	# to be too aggressive about crashing here ...
-	find /opt/couchdb \! \( -user couchdb -group couchdb \) -exec chown -f couchdb:couchdb '{}' +
+	# this is where runtime configuration changes will be written.
+	# we need to explicitly touch it here in case /opt/couchdb/etc has
+	# been mounted as an external volume, in which case it won't exist.
+	# If running as the couchdb user (i.e. container starts as root),
+	# write permissions will be granted below.
+	touch /opt/couchdb/etc/local.d/docker.ini
+
+	# if user is root, assume running under the couchdb user (default)
+	# and ensure it is able to access files and directories that may be mounted externally
+	if [ "$(id -u)" = '0' ]; then
```

Comment:
This is unexpected, can you explain why you're checking for this everywhere? It may not be true, but we may still want the find operation to occur.

/cc @kocolosk

Reply:
Thank you for reviewing @wohali - this line is a leftover from the original [docker-entrypoint.sh:35](https://github.com/apache/couchdb-docker/blob/6a011e4068cc275c896473e011e12e3a087164fa/3.0.0/docker-entrypoint.sh#L35).

Should i review the entire docker-entrypoint script? I do currently not expect too much changes.

Reply:
:thinking: It looks like i somehow polluted my commit - it tells me i actually added that line but i'm pretty sure i didn't.. Investigating... :mag:

Reply:
Yeah, I'm a bit confused here as well. It may just be the diff display. If you can sort this out it'd be appreciated.

Reply:
I guess i based it of from the wrong file version? I have fixed it in minutes.. :smile: